packages/blade/src/components/Popover/_decisions/decisions.md

```diff
@@ -0,0 +1,275 @@
+| headerLeading   | `React.ReactNode`                                                                                                      | `undefined` | Leading element to be placed before header title, recommended components are Icon or Asset. The icon's size will be set to large      |          |
+| footerContent   | `React.ReactNode`                                                                                                      | `undefined` | Slottable content of the footer                                                                                                       |          |
+| content         | `string`                                                                                                               | `undefined` | Content of the popover                                                                                                                |          |
+| children        | `React.ReactNode`                                                                                                      | `undefined` | Trigger component for popover, Accepts any interactive element or icons                                                               | ✅        |
+| placement       | `top, top-start, top-end, left, left-start, left-end, right, right-start, right-end, bottom, bottom-start, bottom-end` | `top`       | Placement of popver, the popver avoid collision with the edge of the screen and flip to the oppisite side even when placement is set. |          |
+| isOpen          | `boolean`                                                                                                              | `undefined` | Controlled state of the popover                                                                                                       |          |
+| defaultIsOpen   | `boolean`                                                                                                              | `undefined` | Uncontrolled state of the popover                                                                                                     |          |
+| onOpenChange    | `({ isOpen }) => void`                                                                                                 | `undefined` | Called when popover isOpen state is changed, this can be used to detect when popover opens or closed                                  |          |
+| initialFocusRef | `React.RefObject`                                                                                                      | `undefined` | The ref of the element that should receive focus when the popover opens.                                                              |          |  |
```

Comment:
onDismiss button click? consumers might need it for triggering some actions or analytics

Reply:
🤔 yeah but right now onOpenChange will be called when close button is clicked or click outside popover, adding onDismissClick might get a bit redundant.

Reply:
there are 2 events that will happen here

onDismiss will also trigger onOpenChange because that's an action user is performing explicitly and onOpenChange is a handler that we are triggering when the state of the tooltip is changing right?

Reply:
Sure, but should onDismiss also be called on clicking outside the popover or just clicking on close button only? 

clicking outside popover to close it is also an user action. 🤔

Reply:
no only explicitly clicking on the button you can call it `onDismissButtonClick` for better context and understanding. outside click to you'll trigger `onOpenChange` na

Reply:
Done.